shell/platform/common/cpp/engine_switches_unittests.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/common/cpp/engine_switches.h"
+
+#include "gtest/gtest.h"
+
+namespace flutter {
+
+namespace {
+// Sets |key=value| in the environment of this process.
+void SetEnvironmentVariable(const char* key, const char* value) {
+#ifdef _WIN32
+  _putenv_s(key, value);
+#else
+  setenv(key, value, 1);
+#endif
+}
+
+// Removes |key| from the environment of this process, if present.
+void ClearEnvironmentVariable(const char* key) {
+#ifdef _WIN32
+  _putenv_s(key, "");
+#else
+  unsetenv(key);
+#endif
+}
+}  // namespace
+
+TEST(FlutterProjectBundle, SwitchesEmpty) {
```

Comment:
Should we have a RELEASE based test that asserts setting a switch gives a zero sized array as well?

Reply:
Good idea; done.

Reply:
This turned out to be a *very* good idea, because the `ifdef` wasn't actually working. I thought `FLUTTER_RELEASE` and associated defines were part of a global config, but actually they are only set in one or two places and thus not present by default unless you happen to have a transient dependency on a low-level target that has it as a `public_config`. Thanks for the suggestion!